### PR TITLE
new trigger strategy

### PR DESCRIPTION
**Pull request by PhantomThief**

### CODE REVIEW

File: src/main/java/com/github/phantomthief/collection/impl/GenericSimpleBufferTriggerBuilder.java [hunk 1]
```diff
@@ -58,11 +59,30 @@
...
+    /**
+     * use {@link #interval(long, TimeUnit)} or {@link #triggerStrategy}
+     */
+    @Deprecated
     public GenericSimpleBufferTriggerBuilder<E, C> on(long interval, TimeUnit unit, long count) {
         builder.on(interval, unit, count);
         return this;
     }
 
+    public GenericSimpleBufferTriggerBuilder<E, C> interval(long interval, TimeUnit unit) {
```
Comment (lixian) [inline src/main/java/com/github/phantomthief/collection/impl/GenericSimpleBufferTriggerBuilder.java]: Set interval twice, 1m, 2m. Then it will be 1m?

Comment (PhantomThief) [inline src/main/java/com/github/phantomthief/collection/impl/GenericSimpleBufferTriggerBuilder.java]: ok. it should check invalid trigger settings and throw exception while building the trigger.

File: src/main/java/com/github/phantomthief/collection/impl/SimpleBufferTrigger.java [hunk 2]
```diff
@@ -174,4 +171,9 @@ private void doConsume() {
     public long getPendingChanges() {
         return counter.get();
     }
+
+    public interface TriggerStrategy {
+
+        boolean check(long lastConsumeTimestamp, long changedCount);
```
Comment (PiouseLeo) [inline src/main/java/com/github/phantomthief/collection/impl/SimpleBufferTrigger.java]: the method name "check" is really ambiguous, use "canTrigger" instead?

Comment (PhantomThief) [inline src/main/java/com/github/phantomthief/collection/impl/SimpleBufferTrigger.java]: ok.

**Pull request merged**